ci: disable core dump checks

dundargoc opened the pull request:
- ci: use the same deps directory location for all jobs

Having separate directory location causes failures to be inconsistent
and ultimately confusing. A common problem is a file with a particular
name is searched for the entire repository, which gives different
results if the dependency directory is inside the neovim directory or
outside of it.

- ci: remove core dump checking from test.yml

There is already a superior core dump check in test/helpers.lua, which
makes this obsolete.

- ci: skip core dump check

The core dump check interferes with CI as it interprets any file with
"core" in it to be a core dump, which is incorrect.

dundargoc committed:
ci: use the same deps directory location for all jobs

Having separate directory location causes failures to be inconsistent
and ultimately confusing. A common problem is a file with a particular
name is searched for the entire repository, which gives different
results if the dependency directory is inside the neovim directory or
outside of it.

dundargoc commented:
It does.

justinmk commented:
>  delete core dump

Not sure what is meant here, but the reason for not deleting the core dump is so that it can be inspected when running tests locally.

dundargoc commented:
I meant removing the `Show core dumps` step in the `posix` job in `.github/workflows/test.yml`. The current implementation prevents setting the `deps` folder in the neovim repo as CI will fail if *any* file has "core" in its name. This in turn was the reason the linux jobs passed and only windows failed in https://github.com/neovim/neovim/pull/22481, when both should have failed.

Information on how to process core dumps is poor and conflicting. After researching a few hours on it without a satisfactory solution, and because I know no one will fix this, I concluded that it's best to remove it as it interferes with other parts of the CI. If anyone is interested in this functionality I suggest they re-implement it properly, rather than dumping the workload of fixing their shoddy craftsmanship onto others.

justinmk commented:
> removing the `Show core dumps` step in the `posix` job in `.github/workflows/test.yml`

Don't we want to see the backtrace of a coredump in CI? Especially for the ASAN job. Or does that happen some other way now?

dundargoc commented:
> Don't we want to see the backtrace of a coredump in CI? Especially for the ASAN job.

In a vacuum, yes. But this interferes with other parts of the CI due to its implementation. After a few hours of trying to fix it and failing, I concluded that the time investment needed to fix it isn't worth it. I don't anticipate anyone will actually fix this, so I'd prefer that this doesn't block parts of the CI that is working.

justinmk commented:
If something is broken and we need to unblock it, sure we can do a temporary solution.

But seeing the backtrace for ASAN job is something that is rather useful, and has been used many times in the past.

dundargoc commented:
Hmm. So it turns out it's not the code in `.github/workflows/test.yml` that is causing interference but `check_cores` in `test/helpers`.

dundargoc commented:
Initial testing indicates that the shell core check does the same thing as the lua core check, but worse.

dundargoc commented:
@justinmk Since the shell core checks seems to be an inferior copy of the lua core check, perhaps we should outright remove it? I suspect this could get resolved quicker in the future if the fixer only has to worry about one core check version.

justinmk commented:
Yes, let's drop any "check cores" logic in shell/CI scripts. And just keep (but disable if necessary) the Lua `check_cores`.

dundargoc committed:
ci: remove core dump checking from test.yml

There is already a superior core dump check in test/helpers.lua, which
makes this obsolete.

dundargoc committed:
ci: skip core dump check

The core dump check interferes with CI as it interprets any file with
"core" in it to be a core dump, which is incorrect.
